### PR TITLE
Add new ARGILLA_HOME_PATH environment variable

### PR DESCRIPTION
**Pull request by jfcalvo**
This PR adds `ARGILLA_HOME_PATH` new environment variable with a default value equal to `~/.argilla`.

After that `ARGILLA_HOME_PATH` is used to define the default value for `ARGILLA_DATABASE_URL`.

I have added these changes to `_const.py` file but maybe we should create a `settings.py` file or something similar.

### CODE REVIEW

**Comment by frascuchon** [comment]
We are using this https://docs.pydantic.dev/usage/settings for managing app settings from environment vars.

I think (need to validate) this is also compatible with the idea of an environment file (json, ini or whatever) containing all the application settings (See https://docs.pydantic.dev/usage/settings/#adding-sources)

So, maybe we can add these new env vars to the common app settings that we have already defined in https://github.com/argilla-io/argilla/blob/develop/src/argilla/server/settings.py#L28

File: src/argilla/_constants.py [hunk 1]
```diff
@@ -13,8 +13,18 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-DEFAULT_MAX_KEYWORD_LENGTH = 128
+import os
+from pathlib import Path
+
+ARGILLA_HOME_PATH = os.getenv("ARGILLA_HOME_PATH", os.path.join(Path.home(), ".argilla"))
+ARGILLA_DATABASE_URL = os.getenv(
```
Comment (frascuchon) [inline src/argilla/_constants.py]: Read my general comment in PR

**Comment by jfcalvo** [comment]
> We are using this https://docs.pydantic.dev/usage/settings for managing app settings from environment vars.
> 
> I think (need to validate) this is also compatible with the idea of an environment file (json, ini or whatever) containing all the application settings (See https://docs.pydantic.dev/usage/settings/#adding-sources)
> 
> So, maybe we can add these new env vars to the common app settings that we have already defined in https://github.com/argilla-io/argilla/blob/develop/src/argilla/server/settings.py#L28

I noticed the `settings.py` file but because the class was named `ApiSettings` I thought it was related with the API alone. Maybe we need to change it to `AppSettings` or something similar because this is not only related with the API.

**Review by frascuchon — APPROVED** [review]
Yeah!

**Pull request merged**